MHVLib/Clients/MHVThingClient.m

```
@@ -162,13 +182,48 @@ - (void)getThingsWithQueries:(MHVThingQueryCollection *)queries
          else
          {
              MHVThingQueryResults *queryResults = [self thingQueryResultsFromResponse:response];
```

I think that we need to check if the queryResults parsed correctly by verifying that it is not nil. If it is nil we should return an error so the caller knows that there is a problem with the response.

---

Added.